base/sharedarray.jl

```diff
@@ -512,8 +512,6 @@ function copy!(S::SharedArray, R::SharedArray)
     return S
 end
 
-complex(S1::SharedArray,S2::SharedArray) = convert(SharedArray, complex(S1.s, S2.s))
```

Comment:
> complex.(S1, S2) would be at the mercy of similars return type here, right? so would not return a SharedArray any more unless we were to go back on similar returning SharedArray ?

IIUC, yes, `complex.(A, B)` for `SharedArray`s `A` and `B` would yield an appropriate `<:Array` rather than `<:SharedArray`. Best to keep this method as a `broadcast` specialization as `@ararslan` suggests, or instead require the more explicit `convert(SharedArray, complex.(A, B))` going forward? (Fused cases will require the latter in any case unless we build a generic `broadcast` for `SharedArray`s.) Thanks!